test/src/utils/vectors/vector_float.test.cpp

```diff
@@ -0,0 +1,461 @@
+* @param lower Lower bound for randomly generated values
+* @param upper Upper bound for randomly generated values
+* @returns random float between given bounds
+*/
+float RandomFloatTestingVector(float lower, float upper)
+{
+  std::srand(time(0));
+  float random = (static_cast<float>(rand())) / static_cast<float>(RAND_MAX);
+  float diff = upper - lower;
+  float r = random * diff;
+  return lower + r;
+}
+
+/**
+ * @brief calculates the magnitude of a vector with elements corresponding with those from argument array
+ * @param vector_T array of 100 float values
+ * @returns magnitude of vector with corresponding values to array
+ */
+double CalculateVectorMagnitude(std::array<float, 100> vector_T)
+{
+    double sum_of_squares = 0;
+    for (int i = 0; i < 100; i++) {
+       sum_of_squares = sum_of_squares + vector_T[i]*vector_T[i];
+    }
+     return std::sqrt(sum_of_squares);
+}
+
+
+// -------------------------------------------------------------------------------------------------
+// Set Up
+// -------------------------------------------------------------------------------------------------
+
+/**
+* @brief Struct used for test fixtures checking class functionality and properties for float values.
+*/
+struct vector_test_float : public ::testing::Test
+{
+   protected:
+    Vector<float, 100> test_vector1_float;
+    Vector<float, 100> test_vector2_float;
+    Vector<float, 100> test_vector_default = Vector<float, 100>(0);
+    float constant = RandomFloatTestingVector(0, 1000);
+    Vector<float, 100> test_vector_constant;
+    Vector<float, 100> test_vector_diff;
+    Vector<float, 100> test_vector_sum;
+    Vector<float, 100> test_vector_mult;
+    Vector<float, 100> test_vector_div;
+    Vector<float, 100> test_vector_sqrt;
+    Vector<double, 100> unit_vector1;
+    Vector<double, 100> unit_vector2;
+
+
+    int dimension = 100;
+    std::array<float, 100> elements1;
+    std::array<float, 100> elements2;
+
+    void SetUp()
+    {
+       // Construct two vectors with random elements and their corresponding unit vectors
+        for (int i = 0; i < dimension; i++) {
+           elements1[i] = RandomFloatTestingVector(0, 1000);
+        }
+        test_vector1_float =  Vector<float, 100>(elements1);
+
+        for (int i = 0; i < dimension; i++) {
+           elements2[i] = RandomFloatTestingVector(0, 1000);
+        }
+        test_vector2_float =  Vector<float, 100>(elements2);
+        unit_vector2 = test_vector1_float.toUnitVector();
+    }
+
+  void TearDown() {}
+};
+
+// -------------------------------------------------------------------------------------------------
+// Vector Constructor Testing
+// -------------------------------------------------------------------------------------------------
+
+/**
+* @brief Tests no-argument constructor and zero-vector properties for a vector with float values.
+*/
+TEST_F(vector_test_float, testVectorNoArgumentConstructor)
```

Comment:
Is test a verb? I might just be incredibly dumb but if it's not a verb. Maybe you should change it to handles or something